wgpu-core/src/device/resource.rs

```diff
@@ -171,6 +171,12 @@ impl std::fmt::Debug for Device {
 impl Drop for Device {
     fn drop(&mut self) {
         resource_log!("Drop {}", self.error_ident());
+
+        let device_lost_closure = self.lock_life().device_lost_closure.take();
+        if let Some(closure) = device_lost_closure {
+            closure.call(DeviceLostReason::Dropped, String::from("Device dropped."));
+        }
+
```

Comment:
thought: I suspect that moving the `device_lost_closure` call to `Drop` will resolve some issues I've seen in CTS indicating that it wasn't expecting to see the device lost callback being invoked.